### PR TITLE
[ASP.NET Core Integration] Check for cancellation before executing action

### PR DESCRIPTION
**Pull request by satvu**
### Issue describing the changes in this PR

partial mitigation for #1929, see #2180 for the long-term fix.

### Pull request checklist

* [x] My changes **do not** require documentation changes
  * [ ] Otherwise: Documentation issue linked to PR
* [ ] My changes **should not** be added to the release notes for the next release
  * [x] Otherwise: I've added my notes to `release_notes.md`
* [x] My changes **do not** need to be backported to a previous version
  * [ ] Otherwise: Backport tracked by issue/PR #issue_or_pr
* [x] I have added all required tests (Unit tests, E2E tests)

Issue created for E2E testing of ASP.NET integration.

### CODE REVIEW

**Comment by jviau** [comment]
@satvu I think this is covering a symptom of the bug, but not the root cause. This approach will avoid us making a call to something that was disposed due to cancellation, but the root issue is that we dispose this object _at all_ before the function execution is complete, regardless of cancellation state.

File: extensions/Worker.Extensions.Http.AspNetCore/src/FunctionsMiddleware/FunctionsHttpProxyingMiddleware.cs [hunk 1]
```diff
@@ -53,7 +53,10 @@ public async Task Invoke(FunctionContext context, FunctionExecutionDelegate next
             {
                 ActionContext actionContext = new ActionContext(httpContext, httpContext.GetRouteData(), new ActionDescriptor());
 
-                await actionResult.ExecuteResultAsync(actionContext);
+                if (!httpContext.RequestAborted.IsCancellationRequested && !context.CancellationToken.IsCancellationRequested)
```
Comment (jviau) [inline extensions/Worker.Extensions.Http.AspNetCore/src/FunctionsMiddleware/FunctionsHttpProxyingMiddleware.cs:56]: Can we add a comment that this is a mitigation and the dispose exception is still technically possible?

**Review by jviau — APPROVED** [review]

**Comment by liliankasem** [comment]
> @satvu I think this is covering a symptom of the bug, but not the root cause. This approach will avoid us making a call to something that was disposed due to cancellation, but the root issue is that we dispose this object _at all_ before the function execution is complete, regardless of cancellation state.
> 
> See my comment [here](https://github.com/Azure/azure-functions-dotnet-worker/issues/1929#issuecomment-1743890718)
> 
> Our cancellation token callbacks have introduced a race condition with function execution, which leads to ObjectDisposedException.

I wonder if this is related to another bug I found last month:

> Sometimes when we send a cancelled invocation with a cancellation request to the worker, the CT being used in the
[BindFunctionInputAsync](https://github.com/Azure/azure-functions-dotnet-worker/blob/2b51979d1c3a93c08c322f788138a1fd2b25f51e/src/DotNetWorker.Core/Context/Features/DefaultFunctionInputBindingFeature.cs#L32) method (DefaultFunctionInputBindingFeature.cs) is already cancelled, this throws a Task cancelled exception leading to a 500. I think this is a race of sorts as it happens every other time but not consistently. I can repro by hitting the a HTTP function endpoint several times in a row.

**Comment by satvu** [comment]
Closing as we are prioritizing a long-term fix and skipping this mitigation.

**Pull request closed without merging**